spec/models/user_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  describe 'default users' do
+    let(:valid_user) { build :user }
+    let(:user) { create :user }
+    let(:second) { create :user }
+    let(:post) { create :post }
+    
+    context 'Complete signup details' do
```

Comment:
I would put all validation specs under a `describe` block (within the mother describe block). I will also do the same for association specs and any instance method specs (please see my comment on `user.rb`):

```ruby
describe "User model" do
  describe "validations" do
    context "complete details" do
      .....
    end
    context "name or email is absent" do
      ....
    end
  end
  describe "associations" do
    context "user that will be deleted has posts" do
      it "also deletes all of that user's posts" do
        ..... 
     end
     context "user to be deleted has comments" do
        it "also deletes all that user's comments" do
         ....
       end  
      end
    end
   # You can write the same context block for ensuring all a user's friendships and friend requests are deleted upon user deletion
   end
   describe "instance methods" do
     # follow same pattern as above
   end
   
   describe "class methods and scopes" do
     # follow same pattern as above
   end
end
```

**Please follow this pattern in your other model specs. This makes our specs understandable and easy to follow and relate to the corresponding models being tested**